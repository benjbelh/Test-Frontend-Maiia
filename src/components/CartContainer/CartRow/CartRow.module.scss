.CartRowContainer {
    display: flex;
    align-items: center;

    .ProductImg {
        width: 20%;
    }

    .ProductInfo {
        padding-left: 20px;
        .ProductName {
            font-size: 14px;
        }
    
        .ProductQuantity {
            color: #757575;
        }
    } 
}