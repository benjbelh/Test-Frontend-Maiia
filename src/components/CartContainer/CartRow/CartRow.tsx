import { IconButton } from '@material-ui/core';
import { RemoveCircleOutline } from '@material-ui/icons';
import React, { FunctionComponent } from 'react';
import styles from './CartRow.module.scss';

type CartRowProps = {
  imgPath: string;
  name: string;
  quantity: number;
  onRemoveClicked: () => void;
};

export const CartRow: FunctionComponent<CartRowProps> = ({
  imgPath = '',
  name = '',
  quantity = 0,
  onRemoveClicked = () => null,
}) => (
  <div className={styles.CartRowContainer}>
    <img src={imgPath} alt="cart-row-img" className={styles.ProductImg} />
    <div className={styles.ProductInfo}>
      <div className={styles.ProductName}>{name}</div>
      <div className={styles.ProductQuantity}>Quantity: {quantity}</div>
    </div>
    <IconButton color="primary" aria-label="remove from shopping cart" onClick={onRemoveClicked}>
      <RemoveCircleOutline />
    </IconButton>
  </div>
);

export default CartRow;
