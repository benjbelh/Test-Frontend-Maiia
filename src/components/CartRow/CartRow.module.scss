.CartRowContainer {
    display: flex;
    align-items: center;
    padding: 8px 0;
    height: 60px;
    

    .ProductImg {
        width: 20%;
    }

    .ProductInfo {
        padding-left: 20px;
        flex-grow: 1;

        .ProductName {
            font-size: 14px;
            display: -webkit-box;
            -webkit-box-orient: vertical;
            -webkit-line-clamp: 2;
            overflow: hidden;
        }
    
        .ProductQuantity {
            color: #757575;
            display: flex;
            align-items: center;

            .QuantityIcon {
                cursor: pointer;
                padding-left: 8px;
            }
        }
    } 
}