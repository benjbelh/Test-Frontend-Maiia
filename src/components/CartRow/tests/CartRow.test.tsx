import React from 'react';
import { render } from '@testing-library/react';
import CartRow from '../CartRow';

describe('CartRow', () => {
  it('should renders correctly with properties', () => {
    const { asFragment } = render(
      <CartRow
        imgPath="https://test.com/img.jpg"
        name="Item 1"
        quantity={0}
        onRemoveProduct={() => null}
        onIncreaseQuantity={() => null}
        onDecreaseQuantity={() => null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
