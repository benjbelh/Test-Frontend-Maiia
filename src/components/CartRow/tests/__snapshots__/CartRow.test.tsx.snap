// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartRow should renders correctly with properties 1`] = `
<DocumentFragment>
  <div
    class="CartRowContainer"
  >
    <img
      alt="cart-row-img"
      class="ProductImg"
      src="https://test.com/img.jpg"
    />
    <div
      class="ProductInfo"
    >
      <div
        class="ProductName"
      >
        Item 1
      </div>
      <div
        class="ProductQuantity"
      >
        Quantity: 0 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root QuantityIcon MuiSvgIcon-colorAction"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root QuantityIcon MuiSvgIcon-colorDisabled"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
          />
        </svg>
      </div>
    </div>
    <button
      aria-label="remove from shopping cart"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
